tests/gradients/core/test_pulse_gradient.py

```diff
@@ -1094,6 +1367,64 @@ def ansatz(params):
         for j0, j1 in zip(jac_bc, jac_no_bc):
             assert qml.math.allclose(j0, j1)
 
+    def test_with_drive_exact(self):
+        """Test that a HardwareHamiltonian only containing a drive is differentiated correctly
+        for a constant amplitude and zero frequency and phase."""
+        import jax
+
+        # values from https://arxiv.org/pdf/2203.06818.pdf
+        timespan = 0.4
+
+        H = qml.pulse.transmon_drive(qml.pulse.constant, 0.0, 0.0, wires=[0])
+        atol = 1e-5
+        dev = qml.device("default.qubit.jax", wires=1)
+
+        def cost(params):
+            qml.evolve(H, atol=atol)(params, t=timespan)
+            return qml.expval(qml.PauliZ(0))
+
+        cost = qml.QNode(cost, dev, interface="jax", diff_method=qml.gradients.stoch_pulse_grad)
+        cost_jax = qml.QNode(cost, dev, interface="jax")
+        params = (0.42,)
+
+        gradfn = jax.grad(cost)
+        res = gradfn(params)
+        exact = jax.grad(cost_jax)(params)
+        assert qml.math.allclose(res, exact)
+
+    def test_with_drive_approx(self):
+        """Test that a HardwareHamiltonian only containing a drive is differentiated
+        approximately correctly for a constant phase and zero frequency."""
+        import jax
+
+        # values from https://arxiv.org/pdf/2203.06818.pdf
+        timespan = 0.4
+
+        H = qml.pulse.transmon_drive(0.0, qml.pulse.constant, 0.0, wires=[0])
+        atol = 1e-5
+        dev = qml.device("default.qubit.jax", wires=1)
+
+        def cost(params):
+            qml.evolve(H, atol=atol)(params, t=timespan)
+            return qml.expval(qml.PauliZ(0))
+
+        cost = qml.QNode(
+            cost,
+            dev,
+            interface="jax",
+            diff_method=qml.gradients.stoch_pulse_grad,
+            num_split_times=20,
+            use_broadcasting=True,
+            sampler_seed=4123,
+        )
+        cost_jax = qml.QNode(cost, dev, interface="jax")
+        params = (0.42,)
+
+        gradfn = jax.grad(cost)
+        res = gradfn(params)
+        exact = jax.grad(cost_jax)(params)
+        assert qml.math.allclose(res, exact)
```

Comment:
I think this only works because the result is strictly 0, else I would be surprised that they are that close with just 20 intermediate points (note that even for constant Hamiltonians the integrand is not constant)

Reply:
nice!